### PR TITLE
Add OSC Receivers for Pattern

### PR DESCRIPTION
**Pull request by reno-**

### CODE REVIEW

**Comment by SteffeyDev** [comment]
Thanks! As with the other PR, please rebase against my `develop` branch and remove duplicate commits. Let me know if you need help with the `git` commands needed here. Also, please update README with the new endpoints you have added.

File: atemOSC/OSCReceiver.mm [hunk 1]
```diff
@@ -553,8 +553,180 @@ - (instancetype) initWithDelegate:(AppDelegate *) delegate
...
+		int key = [[d objectForKey:@"<key>"] intValue];
+		[s keyers][me-1][key-1]->ResetMask();
+	}];
+	
+	[self addEndpoint:@"/me/<me>/usk/<key>/pattern/style" label:@"Set USK<key> Pattern Style" valueType:OSCValString handler:^void(Switcher *s, NSDictionary *d, OSCValue *v) {
+		int me = [[d objectForKey:@"<me>"] intValue];
+		int key = [[d objectForKey:@"<key>"] intValue];
+		if ([[v stringValue] isEqualToString:@"bar_left2right"])
+		{
+			if (IBMDSwitcherKeyPatternParameters* patternParams = [weakSelf getUSKPatternParams:key forSwitcher:s andME:me])
```
Comment (SteffeyDev) [inline atemOSC/OSCReceiver.mm:598]: This section of code could be simpler by using an `NSDictionary` to store the mapping of the string value to the `bmd` constant, then indexing the dictionary to get the constant for a given OSC value.

Comment (SteffeyDev) [reply atemOSC/OSCReceiver.mm:598]: The result should be a dictionary definition followed by a single `SetPattern` call if the value exists in the dict, or print a useful message if not

Comment (reno-) [reply atemOSC/OSCReceiver.mm:598]: I'm sorry about that, but I don't know how to do… The only thing I did was copy/paste. make these changes is really out of my skills… I'm sorry

File: README.md [hunk 2]
```diff
@@ -197,6 +197,20 @@ By default, commands will be sent to the first mix effect block (M/E).  To send
  - **Reset Position, Size, & Rotation to defaults for Upstream Keyer $i** `/atem/usk/$i/fly/reset`
  - **Reset Position, Size, & Rotation to full screen with no rotation for Upstream Keyer $i** `/atem/usk/$i/fly/clear`
 
+#### USK Pattern Parameters
+ - **Set Pattern type for Upstream Keyer $i** `/atem/usk/$i/pattern/type <bar_left2right|bar_top2bottom|barndoor_horizontal|barndoor_vertical|corners|iris_rectangle|iris_diamond|iris_circle|box_topleft|box_topright|box_bottomright|box_bottomleft|box_topcenter|box_centerright|box_centerbottom|box_centerleft|diagonal_topleft|diagonal_topright>`
+ - **Set Inverse Pattern Parameter for Upstream Keyer $i** `/atem/usk/$i/pattern/inverse <bool>`
+- **Set Pattern Size Parameter for Upstream Keyer $i** `/atem/usk/$i/pattern/size <float>`
```
Comment (SteffeyDev) [inline README.md:203]: Indentation of `-` is different between this line and the line before

File: atemOSC/OSCReceiver.mm (hunk 1 above)
Comment (SteffeyDev) [reply atemOSC/OSCReceiver.mm:598]: That's fine, I can refactor after the merge

**Comment by SteffeyDev** [comment]
Looks like this PR still contains some extra commits, both of mine and from the other PR.  Please use `git rebase -i` against my develop branch and `drop` and extra commits.

**Comment by reno-** [comment]
> Looks like this PR still contains some extra commits, both of mine and from the other PR. Please use `git rebase -i` against my develop branch and `drop` and extra commits.

i'm sorry I don't understand which commits I must drop…

**Pull request closed without merging**